presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -101,6 +104,10 @@
     protected final String connectionUrl;
     protected final Properties connectionProperties;
     protected final String identifierQuote;
+    private final boolean mapTableNames;
+
+    private Map<String, Map<String, String>> schemaTableMapping = new HashMap<>();
```

Comment:
Also, in response about the lack of tests, if you read the 2nd part of my initial commit comment:

> A test was attempted to be written similar to the TestJdbcClient with a new TestingDatabase, however it was discovered that H2 will return TRUE for metadata.storesUpperCaseIdentifiers() even if DATABASE_TO_UPPER=FALSE is set. The only way to have that return FALSE is using MODE=MYSQL, however in that case H2 lowercases all table names. So it is not possible to create a test with H2 without modifying the section of code with the call to metadata.storesUpperCaseIdentifiers().